### PR TITLE
[mdns] fix typos

### PR DESCRIPTION
**Pull request by jrhodie**

### CODE REVIEW

**Comment by abtink** [comment]
Thanks @jrhodie. I added typo fixes in https://github.com/openthread/openthread/pull/9954.

File: include/openthread/mdns.h [hunk 1]
```diff
@@ -335,7 +335,7 @@ otError otMdnsUnregisterService(otInstance *aInstance, const otMdnsService *aSer
  *
  * The fields in @p aKey follow these rules:
  *
- * - If the key is associated with a host entry, `mName` specifies the host name and `mServcieType` MUST be NULL.
+ * - If the key is associated with a host entry, `mName` specifies the host name and `mServiceType` MUST be NULL.
  * - If the key is associated with a service entry, `mName` specifies the service instance label (always treated as
  *   a single label) and `mServiceType` specifies the service type (e.g., "_tst._udp"). In this case the DNS name for
  *   key record is `<mName>.<mServiceTye>`.
```
Comment (Vyrastas) [inline include/openthread/mdns.h:341]: Actually looks like this should be fixed too?

mServiceTye -> mServiceType

**Review by abtink — APPROVED** [review]

File: include/openthread/mdns.h (hunk 1 above)
Comment (jrhodie) [reply include/openthread/mdns.h]: good catch, fixed these now too.

**Comment by Vyrastas** [comment]
> Thanks @jrhodie. I added typo fixes in #9954.

@abtink Do we need this PR then, if you're fixing in the other?

**Comment by abtink** [comment]
> > Thanks @jrhodie. I added typo fixes in #9954.
> 
> @abtink Do we need this PR then, if you're fixing in the other?

We can close this one. I fixed them all in #9954.
There were more instances of the same typo in other files  (in the comments in `cpp` file and variable names). I searched and replaced them all in the code.

**Pull request closed without merging**